### PR TITLE
throw PNSE for unsupported SSL options in Quic.

### LINKED ISSUE #55759: [QUIC] Throw PNSE for unsupported SslOptions (client and server)
Todo: please at least make a list which one are not supported.

Related https://github.com/dotnet/runtime/issues/32069 and https://github.com/dotnet/runtime/issues/32071

### PR DESCRIPTION
**Pull request by wfurt**
throw for few options we do not support to make it obvious

fixes #55759

### CODE REVIEW

**Comment by am11** [comment]
>  throw PNSP for unsupported SSL options in Quic.

PNSP -> PNSE?

**Review by ManickaP — COMMENTED** [review]
Where would the check be for #55468?

**Comment by wfurt** [comment]
> Where would the check be for #55468?

I don't understand the comment @ManickaP. I assume  #55468 would go in so I did not add check `ServerCertificateSelectionCallback` just to remove it later. If you had something else in mind let me know.

**Comment by ManickaP** [comment]
You've added the validation into `Create` methods, which is great and makes sense here. In that PR you're adding a new `Create` method, which is not called when the listener is initialized, but later in the game. So I'm asking, how will the validation look like in that case? When and where will it happen once that PR is in?

**Comment by wfurt** [comment]
ok, that make sense. I will update  #55468 as well since I'm going to probably touch it anyway. We may split pre-validation to separate method but I'm not sure I want to do that right now. ConfigurationOpenDelegate may still fail so we should figure out how to better surface exceptions.

File: src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs [hunk 1]
```diff
@@ -36,20 +36,39 @@ protected override bool ReleaseHandle()
...
         {
             X509Certificate? certificate = null;
-            if (options.ClientAuthenticationOptions?.ClientCertificates != null)
+
+            if (options.ClientAuthenticationOptions != null)
             {
-                foreach (var cert in options.ClientAuthenticationOptions.ClientCertificates)
+                if (options.ClientAuthenticationOptions.CipherSuitesPolicy != null)
+                {
+                    throw new  PlatformNotSupportedException(SR.Format(SR.net_quic_ssl_option, nameof(options.ClientAuthenticationOptions.CipherSuitesPolicy)));
```
Comment (ManickaP) [inline src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/SafeMsQuicConfigurationHandle.cs:44]: NIT:
```suggestion
                    throw new PlatformNotSupportedException(SR.Format(SR.net_quic_ssl_option, nameof(options.ClientAuthenticationOptions.CipherSuitesPolicy)));
```
All the throws have double space.

**Review by ManickaP — APPROVED** [review]
LGTM, thanks. Looking forward for the SNI PR solution.

**Comment by wfurt** [comment]
failing test fixed by #56047.

**Pull request merged**